lib/utils.js

```diff
@@ -16,29 +16,74 @@ function getServerlessConfigFile() {
   throw new Error('Cannot find serverless config');
 }
 
-function iterateOverServices(serverless, options, func) {
+function promiseSeries(array, fn) {
+  return array.reduce((promise, value) => {
+    return promise.then(() => fn(value));
+  }, Promise.resolve());
+}
+
+function promiseSeriesMap(array, fn) {
+  const results = [];
+  return promiseSeries(array, (v) => {
+    return fn(v).then(res => results.push(res));
+  }).then(() => results);
+}
+
+function getServiceFolders(serverless, options) {
   const services = options.service ? options.service.split(',') : undefined;
   const workingDir = process.cwd();
   const targetFolder = path.resolve(workingDir, _.get(serverless, 'service.custom.multi.location', 'services'));
   let contents;
   try {
     contents = fs.readdirSync(targetFolder).map(file => path.join(targetFolder, file));
   } catch (e) {
-    return Promise.resolve();
+    throw new Error(`Target folder ${targetFolder} does not exist.`);
   }
-  const serviceFolders = contents.filter(file => isDirectory(file) && fs.existsSync(path.join(file, '/service.yml')));
-  return serviceFolders.reduce((promise, folder) => {
-    const serviceName = path.basename(folder);
-    // should throw error if serviceFolders do not include service specified in options
-    if (services && !services.includes(serviceName)) return promise;
+  const serviceFolders = contents.filter((file) => {
+    if (!isDirectory(file)) return false;
+    if (!fs.existsSync(path.join(file, '/service.yml'))) return false;
+    const serviceName = path.basename(file);
+    if (services && !services.includes(serviceName)) return false;
+    return true;
+  });
+  // should throw error if services is > service folders
+  return serviceFolders;
+}
+
+function iterateOverServices(serverless, options, func) {
+  const workingDir = process.cwd();
+  let serviceFolders;
+  try {
+    serviceFolders = getServiceFolders(serverless, options);
+  } catch (e) {
+    if (e.message.startsWith('Target folder')) return Promise.resolve();
```

Comment:
Maybe a comment here as to why this error is not thrown but resolved?

Reply:
Cool easy!